src/app/public/modules/tabs/tabset.component.ts

```diff
@@ -130,247 +168,310 @@ export class SkyTabsetComponent
-      return;
-    }
+  private updateTabButtons(activeIndex: SkyTabIndex): void {
+    this.tabButtons.forEach(button => {
+      button.active = this.tabsetService.tabIndexesEqual(button.tabIndex, activeIndex);
+    });
+  }
 
-    const value = params[this.permalinkId];
+  /**
+   * Updates the UI and state of the tabset component after the tab index or
+   * tab components have changed.
+   * @param activeIndex The currently active tab index.
+   * @param regenerateTabButtons Indicates if tab button view models should be regenerated.
+   * Setting this value to `false` will simply update the existing tab buttons. Setting this value
+   * to `true` is only necessary when the underlying tab components have changed and the tab
+   * buttons must reflect those changes.
+   */
+  private updateTabsetComponent(
+    activeIndex: SkyTabIndex,
+    regenerateTabButtons = false
+  ): void {
+    // Let the tabset render before setting the active index.
+    setTimeout(() => {
 
-    let index: number | string;
+      // Activate/deactivate tab components.
+      this.tabs.forEach(tab => {
+        this.tabsetService.tabIndexesEqual(tab.tabIndex, activeIndex)
+          ? tab.activate()
+          : tab.deactivate();
+      });
 
-    this.tabs.forEach((tabComponent, i) => {
-      if (tabComponent.permalinkValue === value) {
-        index = tabComponent.tabIndex;
+      // Update the tab button models.
+      if (regenerateTabButtons || !this.tabButtons.length) {
+        this.tabButtons = this.createTabButtons(activeIndex);
+      } else {
+        this.updateTabButtons(activeIndex);
       }
-    });
-
-    // Only set the active tab if an index was found.
-    if (index !== undefined) {
-      this.tabsetService.activateTabIndex(index);
-    }
-  }
 
-  private setPathParamPermalinkValue(value: string | null): void {
-    const params = this.getPathParams();
+      // Update the dropdown trigger button text.
+      this.dropdownTriggerButtonText = this.tabButtons.find(b => {
+        return this.tabsetService.tabIndexesEqual(b.tabIndex, activeIndex);
+      })?.buttonText;
 
-    if (value === null) {
-      delete params[this.permalinkId];
-    } else {
-      params[this.permalinkId] = value;
-    }
+      // Set the query params based on active tab.
+      if (this.permalinkId) {
+        const activeTabComponent = this.tabs.find(tab => {
+          return this.tabsetService.tabIndexesEqual(tab.tabIndex, activeIndex);
+        });
+        this.permalinkService.setParam(this.permalinkId, activeTabComponent?.permalinkValue);
+      }
 
-    // Update the URL without triggering a navigation state change.
-    // See: https://stackoverflow.com/a/46486677
-    const url = this.router.createUrlTree(['.'], {
-      relativeTo: this.activatedRoute,
-      queryParams: params,
-      queryParamsHandling: 'merge'
-    }).toString();
+      // Wait for tab button view models to render before gauging dimensions.
+      setTimeout(() => {
```

Comment:
Maybe not a big deal, but this is `setTimeout` is nested in another `setTimeout`. Is there another more reliable way to wait for the view models to render before detecting overflow?

Reply:
Turns out I only needed the first `setTimeout` on initialization, so I can clean it up a bit.